modules/using-the-build-time-network-policy-generator.adoc

```diff
@@ -0,0 +1,79 @@
+// Module included in the following assemblies:
+//
+// * cli/generating-build-time-network-policies.adoc
+:_content-type: PROCEDURE
+[id="using-the-build-time-network-policy-generator_{context}"]
+= Using the build-time network policy generator
+
+:FeatureName: Build-time network policy generation
+include::snippets/technology-preview.adoc[]
+
+.Prerequisites
+
+. The build-time network policy generator recursively scans the directory you specify when you run the command.
+Therefore, before you run the command, you must already have service manifests, config maps, and workload manifests such as `Pod`, `Deployment`, `ReplicaSet`, `Job`, `DaemonSet`, and `StatefulSet` as YAML files in the specified directory.
+. Verify that you can apply these YAML files as-is using the `kubectl apply -f` command. The build-time network policy generator does not work with files that use Helm-style templating.
+. Verify that the service network addresses are not hardcoded. Every workload that needs to connect to a service must specify the service network address as a variable. You can specify this variable by using the workload's resource environment variable or in a config map.
+
+** link:https://github.com/np-guard/cluster-topology-analyzer/blob/main/tests/k8s_guestbook/frontend-deployment.yaml#L25:L28[Example 1: using an environment variable]
+** link:https://github.com/np-guard/cluster-topology-analyzer/blob/main/tests/onlineboutique/kubernetes-manifests.yaml#L105:L109[Example 2: using a config map]
+** link:https://github.com/np-guard/cluster-topology-analyzer/blob/main/tests/onlineboutique/kubernetes-manifests.yaml#L269:L271[Example 3: using a config map]
+. Service network addresses must match the following official regular expression pattern:
++
+----
+(http(s)?://)?<svc>(.<ns>(.svc.cluster.local)?)?(:<portNum>)? <1>
+----
+<1> In this pattern,
+* <svc> is the service name.
+* <ns> is the namespace where you defined the service.
+* <portNum> is the exposed service port number.
+
++
+Following are some examples that match the pattern:
+* `wordpress-mysql:3306`
+* `redis-follower.redis.svc.cluster.local:6379`
+* `redis-leader.redis`
+* `+http://rating-service.+`
+
+
+.Procedure
+. Verify that the build-time network policy generation feature is available by running the help command:
++
+[source,terminal]
+----
+$ roxctl generate netpol -h
+----
+. Generate the policies by using the `generate netpol` command:
++
+[source,terminal]
+----
+$ roxctl generate netpol <folder-path> <1>
+----
+<1> Specify the path of the folder that has the Kubernetes manifests.
+
+The `roxctl generate netpol` command supports the following options:
```

Comment:
Same notion as above regarding the naming change, feel free to ignore if we do this in a separate PR.